### PR TITLE
Measure CPU time used by a worker

### PR DESCRIPTION
**Pull request by laktek**
## What kind of change does this PR introduce?

This PR introduces way to measure CPU time used by user workers and enforce max CPU time use.

Also, includes some refactoring to move all enforcements into a supervisor thread.

### CODE REVIEW

File: crates/base/src/deno_runtime.rs [hunk 1]
```diff
@@ -207,8 +207,8 @@ impl EdgeRuntime {
                 }
             },
             get_error_class_fn: Some(&get_error_class_name),
-            shared_array_buffer_store: None,
-            compiled_wasm_module_store: None,
+            shared_array_buffer_store: Default::default(),
```
Comment (andreespirela) [inline crates/base/src/deno_runtime.rs:210]: Do we need to set them?

**Comment by andreespirela** [comment]
Great work

File: crates/base/src/deno_runtime.rs (hunk 1 above)
Comment (laktek) [reply crates/base/src/deno_runtime.rs:210]: Will enable only the `compiled_wasm_module_store`. It should speed up wasm usages across isolates.

**Review by andreespirela — APPROVED** [review]
LGTM

**Pull request merged**